service/internal/checksum/middleware_compute_input_checksum.go

```diff
@@ -65,6 +65,10 @@ type ComputeInputPayloadChecksum struct {
 	// when used with trailing checksums, and aws-chunked content-encoding.
 	EnableDecodedContentLengthHeader bool
 
+	checksum string
+
+	sha256Checksum string
```

Comment:
more of a "for my understanding" question, but why is sha256 treated differently than the rest of the algorithms for checksum?

Reply:
Same answer listed above, just for payload signing